lib/zipkin-tracer/faraday/zipkin-tracer.rb

```diff
@@ -51,12 +53,31 @@ def trace!(env, trace_id)
         span.record_tag(Trace::BinaryAnnotation::SERVER_ADDRESS, SERVER_ADDRESS_SPECIAL_VALUE, Trace::BinaryAnnotation::Type::BOOL, remote_endpoint)
         span.record(Trace::Annotation::CLIENT_SEND, local_endpoint)
         response = @app.call(env).on_complete do |renv|
-          # record HTTP status code on response
-          span.record_tag(Trace::BinaryAnnotation::STATUS, renv[:status].to_s, Trace::BinaryAnnotation::Type::STRING, local_endpoint)
+          record_response_tags(span, renv[:status].to_s, local_endpoint)
         end
         span.record(Trace::Annotation::CLIENT_RECV, local_endpoint)
       end
       response
+    rescue Net::ReadTimeout => e
+      msg = "Request timed out."
+      span.record_tag(Trace::BinaryAnnotation::ERROR, msg, Trace::BinaryAnnotation::Type::STRING, local_endpoint)
+      raise e
+    rescue Faraday::ConnectionFailed => e
+      msg = "Request connection failed."
+      span.record_tag(Trace::BinaryAnnotation::ERROR, msg, Trace::BinaryAnnotation::Type::STRING, local_endpoint)
+      raise e
+    rescue Faraday::ClientError => e
+      msg = "Generic Faraday client error."
+      span.record_tag(Trace::BinaryAnnotation::ERROR, msg, Trace::BinaryAnnotation::Type::STRING, local_endpoint)
```

Comment:
how about making some "record_error" method where you could do

```ruby
rescue Faraday::ClientError => e 
  record_error(span, "Generic Faraday client error.", local_endpoint)
  raise
end
```

also, you shouldn't need to `raise e`, `raise` is enough to re-raise:

```ruby
def boom
  raise "kaboom!"
end

begin
  begin
    boom
  rescue => first_err
    puts "first rescue: #{first_err.inspect}, #{first_err.object_id}"
    raise
  end
rescue => second_err
  puts "second rescue: #{second_err.inspect}, #{second_err.object_id}"
end

# => first rescue: #<RuntimeError: kaboom!>, 70353196231420
# => second rescue: #<RuntimeError: kaboom!>, 70353196231420
```